### ISSUE TITLE
Mouse stopped working on recent MSYS2

### ISSUE THREAD

**Issue by JosepMaJAZ** [sarah-walker-pcem/pcem#139]
**Describe the bug**
Mouse is not moving on the emulated machine. (Probably ~~wxwidgets'~~ SDL's fault)
I wrote this [on the forums](https://pcem-emulator.co.uk/phpBB3/viewtopic.php?t=3779), but maybe this is a better place

**To Reproduce**

1. Run a machine with an emulated mouse (Doesn't matter if PS/2 or serial)
2. Run an OS/application where you can see the mouse.
3. Try moving the mouse
4. Mouse does not move

**Expected behavior**
Mouse should move.

**Screenshots**

**Emulator configuration**

- Machine: Tried with GA686BX, 430vx, and even on a ami386
- CPU: PII 233, P MMX 166, 386 SX 25
- Graphics/sound cards: Banshee, soundblaster PCI, trident, sounblaster 16
- Installed OS : Windows 98 SE, MS-DOS 6.22 with cuteMouse.

**Host machine**

- OS: Windows 10
- Self build of "dev" branch
- Build with MSYS2 , updated ( GCC 11.3 , wxwidgets 5.0.3 )

**Additional context**
As I say, I believe this has to be a problem with the ~~upsated wxwidgets~~ SDL, not a problem of PCem itself, but maybe the library is being used in an unusual way that causes this problem now.

Edit: just realised this is done with SDL, not with wxwidgets. Also saw this old post that might suggest that we have to tell it that we want to use the relative mode. https://stackoverflow.com/questions/25576438/sdl-getrelativemousestate-strange-behaviour

Previously, verified that it works on Debian 11 with GNOME. (I was able to try it on a virtual machine by disabling the virtual machine mouse autograbbing)

**Comment by JosepMaJAZ** [comment]
Hello again.
 I found the problem and it is a PCEm bug.
The bug is on the wx-ui/wx-sdl2-display-win.c ,which is only used on windows.

https://github.com/sarah-walker-pcem/pcem/blob/f4a1ab9c5193ebeaa5ab7da3b613dc78a5614550/src/wx-ui/wx-sdl2-display-win.c#L779

This code is intended to center the cursor on the pcem window when it is captured in order to avoid reaching screen borders. 
This could be some legacy code. I see it hasn't changed since 5 years ago (formatting aside).
Basically, SDL is already responsible of maintaining the cursor captured and working in relative changes.

If we want to keep it, the three lines of the mousecapture block should be moved to the window_doinputgrab block, just before the setwindowgrab()  and getrelativemousestate moved to the end.
